crates/topos-clock/src/time.rs

```diff
@@ -0,0 +1,99 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use chrono::{DateTime, Utc};
+use tokio::{
+    spawn,
+    sync::broadcast,
+    time::{interval_at, Instant},
+};
+use tracing::info;
+
+use crate::{Clock, Error, Event};
+
+pub struct TimeClock {
```

Comment:
Agreed. Question about the name: are we expecting further implementations of the `Clock` trait, so that "TimeClock" is but one implementation of it? Can you elaborate a bit on what other `Clock` implementations we anticipate?